src/c3chart/Common.js

```diff
@@ -91,10 +99,55 @@
     Common.prototype.update = function (domNode, element) {
         HTMLWidget.prototype.update.apply(this, arguments);
 
+        if (this.showLegend()) {
+            this.c3Chart.legend.show();
+        } else {
+            this.c3Chart.legend.hide();
+        }
+
         this.c3Chart.resize({
             width: this.width(),
             height: this.height()
         });
+
+        this.c3Chart.load(this._c3LoadDataObj);
+
+        var chartOptions = this.getChartOptions();
```

Comment:
the getChartOptions should be all the options that go into the "load" - the styles will get applied in the normal way within the update method..